src/object/object_primitive.c

```diff
@@ -17577,50 +17579,126 @@ mr_data_cmpdisk_json (void *mem1, void *mem2, TP_DOMAIN * domain, int do_coercio
       first += OR_BYTE_SIZE;
       second += OR_BYTE_SIZE;
 
-      return (strcmp (first, second) == 0) ? DB_EQ : DB_UNK;
+      first_json_body = db_private_strdup (NULL, first);
```

Comment:
1. Too much memory duplication. In normal circumstance we shouldn't have to duplicate bodies, but I know we have to clear document and body is automatically cleared. I think we should add the possibility of clearing body, document and schema separately. In a different issue, not in this patch.
2. If first or second is compressed, both strings are uncompressed. I think we have to treat each separately.